File: algorithms/1_two_pointers/2_sum_square_numbers/sum_square.py
def judgeSquareSum(c):
    big = int(c**(1/2))
    small = 0
    while small<=big:
        if small*small + big*big == c:
            return True
        elif small**2 + big**2 >c:
            big = big-1
        else: small = small +1
    return False


if __name__ == "__main__":
    print(judgeSquareSum(5))
    print(judgeSquareSum(3))

File: algorithms/1_two_pointers/2_sum_square_numbers/sum_square.js
var judgeSquareSum = function(c) {
  let big = Math.floor(c**(1/2));
  let small = 0;
  while (small<=big){
    if (small*small+big*big===c) return true;
    else if (small**2+big**2 >c) big-=1;
    else small +=1;
  }
  return false;
};

module.exports = judgeSquareSum;
